src/V3Width.cpp

```diff
@@ -2094,16 +2094,20 @@ class WidthVisitor final : public VNVisitor {
         // We can't skip this step when width()!=0, as creating a AstVar
         // with non-constant range gets size 1, not size 0.  So use didWidth().
         if (nodep->didWidth()) return;
-        if (nodep->doingWidth()) {  // Early exit if have circular parameter definition
-            UASSERT_OBJ(nodep->valuep(), nodep, "circular, but without value");
+        nodep->doingWidth(true);
+
+        // Check for recursive initialization
+        std::function<bool(AstNode*)> isRecursive = [&](AstNode* np) -> bool {
+            return np && np->exists([&](AstNodeVarRef* refp) {
+                return refp->varp() == nodep || isRecursive(refp->varp()->valuep());
+            });
+        };
```

Comment:
This is needed because of the assertions: you can't call iterateAndNext on the same node twice (recursively). Doing so was never really safe to do due to the later call stomping m_iterpp hence edits in the higher call would have been borked.